lnd.go

```diff
@@ -1105,6 +1114,7 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 			RecoveryWindow: recoveryWindow,
 			Wallet:         newWallet,
 			ChansToRestore: initMsg.ChanBackups,
+			UnloadWallet:   loader.UnloadWallet,
```

Comment:
I suspect that this will not be set if `--noseedbackup` is used which may eventually lead to a crash in the defer function

Reply:
Good thinking. Right now, the defer happens inside this if statement:
https://github.com/lightningnetwork/lnd/blob/97cfbecd4b8971ea87a34e8b165602caac3c4b8f/lnd.go#L375

I don't know how `--noseedbackup` works, but it's still opening a wallet? With the current code, the wallet file will not be unloaded if `--noseedbackup` is used.

Reply:
Yes, indeed it's inside the if statement, missed that. Won't crash.